packages/shared/src/components/feeds/FeedContainer.tsx

```diff
@@ -107,10 +113,30 @@ export const FeedContainer = ({
     insaneMode: listMode,
     loadedSettings,
   } = useContext(SettingsContext);
+  const { user, isLoggedIn } = useAuthContext();
   const { shouldUseFeedLayoutV1 } = useFeedLayout();
   const { feedName } = useActiveFeedNameContext();
   const router = useRouter();
   const onboardingV4dot5 = useFeature(feature.onboardingV4dot5);
+  const userAcquisitionVersion = useFeature(feature.userAcquisition);
+  const showAcquisitionForm =
+    isLoggedIn &&
+    feedName === SharedFeedPage.MyFeed &&
+    userAcquisitionVersion === UserAcquisition.V1 &&
+    (router.query?.[acquisitionKey] as string)?.toLocaleLowerCase() ===
+      'true' &&
+    !user?.acquisitionChannel;
+  const childItems = useMemo(() => {
```

Comment:
Should we consider handling the logic in useFeed rather?
I'm coming from a point of the other epic where we also need to plug into this spot so would be great if it's uniform already.

That also handles memo etc already and could pretty much avoid even adding the first ad for instance. (maybe even not loading on first page? (bonus))

Reply:
I was considering this as well and the implementation is now in a state, where we can easily morph it that way. Let me know if you would prefer that.

Reply:
I went ahead and moved it to `useFeed` hook, it was quite simple to do now. Thanks for the suggestion 👍